### PR TITLE
feat(slide schema)[OK-21] Add schema for slide & reorganize code

### PR DESCRIPTION
**Pull request by marcepanowyy**
Add schema for slide,
Reorganize code structure,
Add slide module,
Modify board schema

### CODE REVIEW

File: backend/src/enums/user.auth.provider.ts [hunk 1]
```diff
@@ -1,5 +1,5 @@
 export enum UserAuthProvider {
-  INTERNAL = 'internal',
-  GOOGLE = 'google',
-  GITHUB = 'github',
+  INTERNAL = 1,
+  GOOGLE = 2,
+  GITHUB = 3,
```
Comment (Wajktor13) [inline backend/src/enums/user.auth.provider.ts:4]: Whats the reason of this change?

File: backend/src/enums/user.role.ts [hunk 2]
```diff
@@ -1,4 +1,10 @@
 export enum UserRole {
-  ADMIN = 'admin',
-  USER = 'user',
+  // can change roles, delete users, etc.
+  ADMIN = 100,
+
+  // can delete users, create/delete/modify boards, etc.
+  MEMBER = 10,
```
Comment (Wajktor13) [inline backend/src/enums/user.role.ts:6]: I don't like this name at all, maybe `MODERATOR`?

**Review by Wajktor13 — APPROVED** [review]

**Review by Wajktor13 — CHANGES_REQUESTED** [review]
.

File: backend/src/enums/user.role.ts (hunk 2 above)
Comment (Wajktor13) [inline backend/src/enums/user.role.ts]: I don't like this name at all, maybe `MODERATOR`?

**Review by Wajktor13 — CHANGES_REQUESTED** [review]

File: backend/src/enums/user.role.ts (hunk 2 above)
Comment (Wajktor13) [reply backend/src/enums/user.role.ts]: I'm also not sure if we need this role

File: backend/src/enums/user.auth.provider.ts (hunk 1 above)
Comment (marcepanowyy) [reply backend/src/enums/user.auth.provider.ts:4]: imo its better to store numbers than strings in db

File: backend/src/enums/user.role.ts (hunk 2 above)
Comment (marcepanowyy) [reply backend/src/enums/user.role.ts]: i'll delete this role

**Review by Wajktor13 — APPROVED** [review]

**Pull request merged**